app/controllers/settings/main_controller.rb

```diff
@@ -67,12 +70,35 @@ def user_params
         :purchasing_power_parity_limit,
         :purchasing_power_parity_payment_verification_disabled,
         :show_nsfw_products,
+        { reply_to_emails: [:id, :email, { product_ids: [] }] },
         { seller_refund_policy: [:max_refund_period_in_days, :fine_print] }
       ]
 
       params.require(:user).permit(permitted_params)
     end
 
+    def upsert_reply_to_emails(reply_to_emails_data)
+      existing_reply_to_emails = current_seller.reply_to_emails
+
+      keep_reply_to_emails = reply_to_emails_data&.map do |email_data|
+        reply_to_email = current_seller.reply_to_emails.find_or_initialize_by(id: email_data[:id])
+        reply_to_email.update!(email: email_data[:email]) if email_data[:email] != reply_to_email.email
+
+        if reply_to_email.persisted?
+          reply_to_email.products.clear
+        end
+
+        if email_data[:product_ids].present?
+          products = current_seller.products.by_external_ids(email_data[:product_ids])
+          reply_to_email.products = products
+        end
+
+        reply_to_email
+      end || []
+
+      (existing_reply_to_emails - keep_reply_to_emails).each(&:destroy)
```

Comment:
I think this would all be much cleaner without a new table and just updating each product.reply_to_email

Reply:
I think that's quite different as it's a one-to-many versus a one-to-one.